units/commit_queue.go

```diff
@@ -476,3 +493,65 @@ func validateBranch(branch *github.Branch) error {
 	}
 	return nil
 }
+
+func addMergeTaskAndVariant(patchDoc *patch.Patch, project *model.Project) error {
+	settings, err := evergreen.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, "error retrieving Evergreen config")
+	}
+
+	mergeBuildVariant := model.BuildVariant{
+		Name:        "commit-queue-merge",
+		DisplayName: "Commit Queue Merge",
+		RunOn:       []string{settings.CommitQueue.MergeTaskDistro},
+		Tasks: []model.BuildVariantTaskUnit{
+			{Name: "merge-patch"},
+		},
+	}
+
+	mergeTask := model.ProjectTask{
+		Name: "merge-patch",
+		Commands: []model.PluginCommandConf{
+			{
+				Command: "git.get_project",
+				Type:    evergreen.CommandTypeSetup,
+				Params: map[string]interface{}{
+					"directory": "${workdir}/src",
+				},
+			},
+			{
+				Command: "git.push",
+				Params: map[string]interface{}{
+					"directory":       "${workdir}/src",
+					"committer_name":  settings.CommitQueue.CommitterName,
+					"committer_email": settings.CommitQueue.CommitterEmail,
+				},
+			},
+		},
+		DependsOn: []model.TaskUnitDependency{
+			{
+				Name:    "*",
+				Variant: "*",
+			},
+		},
+	}
+
+	project.BuildVariants = append(project.BuildVariants, mergeBuildVariant)
+	project.Tasks = append(project.Tasks, mergeTask)
+
+	validationErrors := validator.CheckProjectSemantics(project)
```

Comment:
What is it that we need to validate here that hasn't been done already?

Reply:
Meant to do CheckProjectSyntax (updated now) to run `validateProjectTaskNames` and `validateBVNames` to check the project didn't already have a task/bv with the same name.

Reply:
It also makes sure the distro configured for the task to run on actually exists.